torch_xla/experimental/custom_kernel.py

```diff
@@ -927,6 +935,12 @@ def _ragged_paged_attention_nonkernel(
                                               head_dim)[:kv_len]
     v = kv_pages[indices, :, 1::2, :].reshape(-1, num_kv_heads,
                                               head_dim)[:kv_len]
+    if k_scale is not None:
+      k = k.to(torch.float32) * k_scale
```

Comment:
Do you know the reason why we need to do `k.to(torch.float32)`?

Reply:
curious if it's necessary to make the multiplication  k*k_scale in fp32. If so, do you know the reasoning behind it?